api/src/main/java/org/pmiops/workbench/institution/InstitutionUserInstructionsMapper.java

```diff
@@ -0,0 +1,37 @@
+package org.pmiops.workbench.institution;
+
+import org.mapstruct.AfterMapping;
+import org.mapstruct.Context;
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.MappingTarget;
+import org.owasp.html.HtmlPolicyBuilder;
+import org.owasp.html.PolicyFactory;
+import org.pmiops.workbench.db.model.DbInstitutionUserInstructions;
+import org.pmiops.workbench.model.InstitutionUserInstructions;
+
+@Mapper(componentModel = "spring")
+public interface InstitutionUserInstructionsMapper {
+  @Mapping(target = "institutionUserInstructionsId", ignore = true)
+  @Mapping(target = "institutionId", ignore = true) // set by setFields()
+  @Mapping(target = "userInstructions", ignore = true) // set by setFields()
+  DbInstitutionUserInstructions modelToDb(
+      InstitutionUserInstructions modelObject, @Context InstitutionService institutionService);
+
+  @AfterMapping
+  default void setFields(
+      @MappingTarget DbInstitutionUserInstructions target,
+      InstitutionUserInstructions modelObject,
+      @Context InstitutionService institutionService) {
+
+    final long institutionId =
+        institutionService
+            .getDbInstitutionOrThrow(modelObject.getInstitutionShortName())
+            .getInstitutionId();
```

Comment:
will it make sense to have a method that takes in shortName and send institutionID as well in addition to getDbInstitutionOrThrow because thats being used at a lot of places.

Reply:
We do need the full DbInstitution in a few places so we'd need to have both.

So my preferences is to keep it as-is.

Reply:
i meant adding a new method in addition to getDBInstitution